src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -838,12 +823,13 @@ public bool IsDaylightSavingTime()
 
         public static DateTime SpecifyKind(DateTime value, DateTimeKind kind)
         {
-            return new DateTime(value.InternalTicks, kind);
+            if ((uint)kind > (uint)DateTimeKind.Local) ThrowInvalidKind();
+            return new DateTime(value.UTicks | ((ulong)kind << KindShift));
```

Comment:
>return new DateTime(value.UTicks | ((ulong)kind << KindShift)) [](start = 12, length = 62)

this also can cause the constructor to throw when checking against the MaxTicks?

Reply:
ignore this. I noticed the constructor that takes ulong. 

---
In reply to: [565693181](https://github.com/dotnet/runtime/pull/45479#discussion_r565693181) [](ancestors = 565693181)